### PR TITLE
`User::load()` calls should be avoided in classes #546 fix

### PR DESCRIPTION
**Pull request by ghost**
Updated CS sniffer ruleset, run it, replaced User::load with dependency injection. #546

### CODE REVIEW

File: src/Controller/SubscriptionController.php [hunk 1]
```diff
@@ -36,17 +36,31 @@ class SubscriptionController extends ControllerBase {
...
    * @param \Drupal\Core\Messenger\MessengerInterface $messenger
    *   The messenger service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
    */
-  public function __construct(OgAccessInterface $og_access, MessengerInterface $messenger) {
+  public function __construct(OgAccessInterface $og_access, MessengerInterface $messenger, EntityTypeManagerInterface $entity_type_manager) {
```
Comment (pfrenssen) [inline src/Controller/SubscriptionController.php]: In the `SubscriptionController` we don't need to inject the `EntityTypeManager` since we already inherit the `$this->getEntityTypeManager()` method from the base class (`ControllerBase`).

File: src/Controller/SubscriptionController.php [hunk 2]
```diff
@@ -84,7 +99,7 @@ public function subscribe($entity_type_id, EntityInterface $group, OgMembershipT
       throw new AccessDeniedHttpException();
     }
 
-    $user = User::load($this->currentUser()->id());
+    $user = $this->entityTypeManager->load($this->currentUser()->id());
```
Comment (pfrenssen) [inline src/Controller/SubscriptionController.php]: Here instead we can call `$this->getEntityTypeManager()->getStorage('user')->load()`.

**Review by pfrenssen — CHANGES_REQUESTED** [review]
Looking good, but in one case it is not necessary to inject the entity type manager ourselves, since the base class already has a convenient method that allows us to access it.

**Review by MPParsley — APPROVED** [review]
Looks good

**Review by MPParsley — APPROVED** [review]

**Review by MPParsley — APPROVED** [review]

File: phpcs-ruleset.xml.dist [hunk 3]
```diff
@@ -28,6 +28,8 @@
         <exclude name="DrupalPractice.InfoFiles.NamespacedDependency.NonNamespaced" />
         <!-- https://github.com/Gizra/og/issues/546 -->
         <exclude name="DrupalPractice.Objects.GlobalClass.GlobalClass" />
+        <!-- https://github.com/Gizra/og/issues/547 -->
+        <exclude name="DrupalPractice.Objects.GlobalDrupal.GlobalDrupal" />
```
Comment (pfrenssen) [inline phpcs-ruleset.xml.dist]: These two lines are out of scope for this PR. I think they were probably introduced by accident through a merge conflict. Let's remove them.

**Review by pfrenssen — DISMISSED** [review]

File: phpcs-ruleset.xml.dist [hunk 4]
```diff
@@ -28,6 +28,8 @@
         <exclude name="DrupalPractice.InfoFiles.NamespacedDependency.NonNamespaced" />
         <!-- https://github.com/Gizra/og/issues/546 -->
         <exclude name="DrupalPractice.Objects.GlobalClass.GlobalClass" />
```
Comment (MPParsley) [inline phpcs-ruleset.xml.dist]: We should delete these though.

File: phpcs-ruleset.xml.dist (hunk 3 above)
Comment (MPParsley) [reply phpcs-ruleset.xml.dist]: We should delete these lines (they've probably been reintroduced in the merge conflict):
```
        <!-- https://github.com/Gizra/og/issues/546 -->
        <exclude name="DrupalPractice.Objects.GlobalClass.GlobalClass" />

```

**Review by MPParsley — APPROVED** [review]

**Review by pfrenssen — APPROVED** [review]
